pkg/worker/worker_test.go

```diff
@@ -0,0 +1,84 @@
+package worker
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"github.com/stretchr/testify/assert"
+)
+
+var w = NewWorker(conf.NewConfig())
+
+func TestStartExistingCommand(t *testing.T) {
+	jobID, err := w.Start(Command{Name: "ls"})
+
+	assert.NotEmpty(t, jobID)
+	assert.Nil(t, err)
+}
+
+func TestStartNotExistingCommand(t *testing.T) {
+	jobID, err := w.Start(Command{Name: "notexists"})
+
+	assert.NotEmpty(t, jobID)
+	assert.NotNil(t, err)
+}
+
+func TestStopNotExistingProcess(t *testing.T) {
+	err := w.Stop("")
+	assert.Equal(t, "Job not found", err.Error())
+}
+func TestStopExistingProcess(t *testing.T) {
+	jobID, err := w.Start(Command{Name: "sleep", Args: []string{"2"}})
+	assert.NoError(t, err)
+
+	err = w.Stop(jobID)
+	assert.NoError(t, err)
+}
+
+func TestQueryExistingProcess(t *testing.T) {
+	jobID, err := w.Start(Command{Name: "sleep", Args: []string{"1"}})
+	assert.NoError(t, err)
+
+	st, err := w.Query(jobID)
+	assert.False(t, st.Exited)
+	assert.Zero(t, st.ExitCode)
+	assert.NoError(t, err)
+
+	time.Sleep(time.Second * 2)
+
+	st, err = w.Query(jobID)
+	assert.NoError(t, err)
+	assert.True(t, st.Exited)
+	assert.Zero(t, st.ExitCode)
+}
+
+func TestQueryStoppedProcess(t *testing.T) {
+	jobID, err := w.Start(Command{Name: "ls", Args: []string{"1"}})
+	assert.NoError(t, err)
+
+	err = w.Stop(jobID)
+	assert.NoError(t, err)
+
+	time.Sleep(time.Second * 2)
+
+	st, err := w.Query(jobID)
+	assert.NoError(t, err)
+	assert.False(t, st.Exited)
+	assert.Equal(t, -1, st.ExitCode)
```

Comment:
Aren't these the values for a currently running job, not a stopped job?

Reply:
No, the job was forced to stop at line 61.

Reply:
My point is that if the expected values of a stopped job are `exited = false` and `exitCode = -1`, it's indistinguishable from a running job.

Reply:
Nevermind, I see now that exited isn't set until the process exits, I thought it would be false while the process runs. My bad!

Reply:
No problem. Yes, false is the initial job state.